controllers/actions.github.com/ephemeralrunnerset_controller.go

```diff
@@ -290,6 +335,47 @@ func (r *EphemeralRunnerSetReconciler) createEphemeralRunners(ctx context.Contex
 	return multierr.Combine(errs...)
 }
 
+func (r *EphemeralRunnerSetReconciler) createProxySecret(ctx context.Context, ephemeralRunnerSet *v1alpha1.EphemeralRunnerSet, log logr.Logger) error {
+	proxySecretData, err := ephemeralRunnerSet.Spec.EphemeralRunnerSpec.Proxy.ToSecretData(func(s string) (*corev1.Secret, error) {
+		secret := new(corev1.Secret)
+		err := r.Get(ctx, types.NamespacedName{Namespace: ephemeralRunnerSet.Namespace, Name: s}, secret)
+		return secret, err
+	})
+	if err != nil {
+		return fmt.Errorf("failed to convert proxy config to secret data: %w", err)
+	}
+
+	runnerPodProxySecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      proxyEphemeralRunnerSetSecretName(ephemeralRunnerSet),
+			Namespace: ephemeralRunnerSet.Namespace,
+			Labels:    map[string]string{
```

Comment:
do we want to fix this TODO?

Reply:
Thanks, I forgot this was a thing :( never again PRs this size, feel free to quote me on that 😄